arch/src/x86_64/mod.rs

```diff
@@ -802,6 +802,21 @@ pub fn get_host_cpu_phys_bits() -> u8 {
         } else {
             36
         }
+    };
+
+    #[cfg(feature = "tdx")]
+    {
+        // When running TDX guest, the Guest Physical Address space is limited
+        // by a shared bit that is located on bit 47 for 4 level paging, and on
+        // bit 51 for 5 level paging (when GPAW bit is 1). In order to keep
+        // things simple, and since a 47 bits address space is 128TiB large, we
+        // ensure to limit the physical addressable space to 47 bits when
+        // runnning TDX.
+        std::cmp::min(host_phys_bits, 47)
```

Comment:
This should be set if TDX is in use not just if it is compiled in. Because we will remove the tdx feature flag at some point in the near future.

Reply:
I think vm.rs's `physical_bits()` could easily be modified to check the config to see if TDX is in use.

Reply:
Done!